src/common/attachments/uploadAttachmentsDialog.component.tsx

```diff
@@ -0,0 +1,181 @@
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  useTheme,
+} from '@mui/material';
+import AwsS3, { AwsBody } from '@uppy/aws-s3';
+import Uppy, { Meta, UppyFile } from '@uppy/core';
+import '@uppy/core/dist/style.css';
+import '@uppy/dashboard/dist/style.css';
+import ProgressBar from '@uppy/progress-bar'; // Import the ProgressBar plugin
+import { Dashboard } from '@uppy/react';
+import React from 'react';
+import { usePostAttachmentMetadata } from '../../api/attachments';
+
+const MAX_FILE_SIZE_MB = 100;
+const MAX_FILE_SIZE_B = MAX_FILE_SIZE_MB * 1000 * 1000;
+
+export interface UploadAttachmentsDialogProps {
+  open: boolean;
+  onClose: () => void;
+  entityId: string;
+}
+
+const UploadAttachmentsDialog = (props: UploadAttachmentsDialogProps) => {
+  const { open, onClose, entityId } = props;
+  const theme = useTheme();
+
+  const [isUploading, setIsUploading] = React.useState(false);
+  const [uppy, setUppy] = React.useState<Uppy<Meta, AwsBody> | null>(null);
+  const [fileMetadataMap, setFileMetadataMap] = React.useState<
+    Record<string, string>
+  >({});
+
+  const { mutateAsync: postAttachmentMetadata } = usePostAttachmentMetadata();
+
+  // Handlers for upload-error and file-removed events
+  const handleUploadError = React.useCallback(
+    (fileMetadataMap: Record<string, string>) =>
+      (file?: UppyFile<Meta, AwsBody>) => {
+        const id = fileMetadataMap[file?.id ?? ''];
+        // TODO CHeck if it exist in database first
+        // This should fix the multiple deletion error
+        if (id) {
+          // TODO delete if the upload has failed
+          // console.log('Delete id ', id);
+          const newMap = { ...fileMetadataMap };
+          delete newMap[file?.id ?? ''];
+        }
+        setIsUploading(false);
+      },
+    []
+  );
+
+  const handleFileRemoved = React.useCallback(
+    (fileMetadataMap: Record<string, string>) =>
+      (file: UppyFile<Meta, AwsBody>) => {
+        const id = fileMetadataMap[file?.id ?? ''];
+        // TODO CHeck if it exist in database first
+        // This should fix the multiple deletion error
+        if (id) {
+          // TODO delete if the file has been removed mid upload
+          // console.log('Delete id ', id);
+          const newMap = { ...fileMetadataMap };
+          delete newMap[file?.id ?? ''];
+        }
+
+        setIsUploading(false);
+      },
+    []
+  );
+
+  const handleClose = React.useCallback(() => {
+    onClose();
+    setFileMetadataMap({});
+    setIsUploading(false);
+  }, [onClose]);
+
+  React.useEffect(() => {
+    if (open) {
+      const uppyInstance = new Uppy<Meta, AwsBody>({
+        autoProceed: false,
+        restrictions: {
+          maxFileSize: MAX_FILE_SIZE_B,
+          requiredMetaFields: ['name'],
+        },
```

Comment:
Could this not be done outside the use effect so it doesn't keep initialising unnecessarily? (There are several examples on https://uppy.io/docs/react/)

Reply:
It doesn't initialising unnecessarily but i can refactor the code so look similar to that. If i moved it outside it renders too much loool

Reply:
Yeah I think it would be worth aligning it with their examples. In this case as its in the useEffect every time the `open`, `entityId` or `postAttachmentMetadata` changes it creates a `new Uppy<Meta, AwsBody>` which reinitialises it.

Reply:
Yeah I think it would be worth aligning it with their examples. In this case as its in the useEffect every time the `open`, `entityId` or `postAttachmentMetadata` changes it creates a `new Uppy<Meta, AwsBody>` which reinitialises it. Not sure whether it can be completely eliminated though, I guess it depends on what is needed in the `use`. Their last example on the page would work I think, but then it couldn't use react query.